### ISSUE TITLE
Null Referance Exception when loading referance property

### ISSUE THREAD

**Issue by gokhanabatay** [nhibernate/nhibernate-core#2191]
Why owner is null, if it calls nullable referance property then it gets null referance exception.
"resultRow[i] = ResultTypes[i].NullSafeGet(rs, scalarColumns[i], session, **null**);"

15.04.2017 NHibernate Code issues this error sorry, latest release does not this error.

QueryLoader.cs

```c#
protected override object[] GetResultRow(object[] row, DbDataReader rs, ISessionImplementor session)
{
	object[] resultRow;

	if (_hasScalars)
	{
		string[][] scalarColumns = _scalarColumnNames;
		int queryCols = ResultTypes.Length;
		resultRow = new object[queryCols];
		for (int i = 0; i < queryCols; i++)
		{
			resultRow[i] = ResultTypes[i].NullSafeGet(rs, scalarColumns[i], session, null);
		}
	}
	else
	{
		resultRow = ToResultRow(row);
	}

	return resultRow;
}
```

ManyToOneType.cs below entry is null because owner is null so it gets null referance exception

```c#
public override bool IsNull(object owner, ISessionImplementor session)
{
	if (IsNullable && !string.IsNullOrEmpty(PropertyName))
	{
		EntityEntry entry = session.PersistenceContext.GetEntry(owner);

		return session.PersistenceContext.IsPropertyNull(entry.EntityKey, PropertyName);
	}

	return false;
}
```

My below sample is complex but above code shows when owner is null error occours.
```c#
var stmtInfos =
               (from ctd in ISession.Query<StmtSummary>()
                where
                ctd.Key.StmtDate == SqlNumberDate.DefaultDate
                select new
                {
                    StmtSummary = ctd.LastStmtSummary == null ? ctd : ctd.LastStmtSummary
		}).ToList();
//Tupple Generic Class for overriding GetHashCode and Equals method for NH
public class StmtSummaryKey : Tupple<StmtSummaryKey>, ICloneable
{
        [Required]
        public virtual string AccountNo { get; set; }
        [Required]
        public virtual DateTime? StmtDate { get; set; }
        [Required]
        public virtual short CurrencyCode { get; set; }

        public object Clone()
        {
            return MemberwiseClone();
        }
}

public class StmtSummary 
{
        public virtual StmtSummaryKey Key { get; set; }

        [JsonIgnore]
        public virtual LastStmtSummary LastStmtSummary { get; set; }
}

public class StmtSummaryMap : EntityMap<StmtSummary>
{
        public StmtSummaryMap()
        {
            Schema("");
            Table("");
            CompositeId(x => x.Key).KeyProperty(y => y.AccountNo, z => z.ColumnName("ACCOUNT_NO").Length(10))
                                            .KeyProperty(y => y.StmtDate, z => z.ColumnName("STMT_DATE")).CustomType<SqlNumberDate>()
                                            .KeyProperty(y => y.CurrencyCode, z => z.ColumnName("CURRENCY_CODE"));

            References(x => x.LastStmtSummary).Columns(new[] { "ACCOUNT_NO", "PREV_STMT_DATE", "CURRENCY_CODE" })
                .ReadOnly().LazyLoad().NotFound.Ignore();
}

public class LastStmtSummary : StmtSummary
{

}

public class LastStmtSummaryMap : EntityMap<LastStmtSummary>
{
        public LastStmtSummaryMap()
        {
            Schema("");
            Table("");
            CompositeId(x => x.Key).KeyProperty(y => y.AccountNo, z => z.ColumnName("ACCOUNT_NO").Length(10))
                                            .KeyProperty(y => y.StmtDate, z => z.ColumnName("STMT_DATE")).CustomType<SqlNumberDate>()
                                            .KeyProperty(y => y.CurrencyCode, z => z.ColumnName("CURRENCY_CODE"));

            References(x => x.LastStmtSummary).Columns(new[] { "ACCOUNT_NO", "PREV_STMT_DATE", "CURRENCY_CODE" })
                .ReadOnly().LazyLoad().NotFound.Ignore();
            Where("STMT_DATE != 19000101");
	    Polymorphism.Explicit();
}
```

**Comment by bahusoid** [comment]
I believe this is another case that should be fixed by #2082

**Comment by bahusoid** [comment]
Closing due to author comment:
> 15.04.2017 NHibernate Code issues this error sorry, latest release does not this error
